src/labels.js

```diff
@@ -0,0 +1,443 @@
+    let major;
+    let minor;
+    let patch;
+
+    if (object && object.version) {
+      major = parseInt(object.version.split('.')[0]);
+      minor = parseInt(object.version.split('.')[1]);
+      patch = parseInt(object.version.split('.')[2]);
+    }
+
+    // First time, migrate from wallet payload if needed
+    if (object === null || Helpers.isEmptyObject(object)) {
+      if (this._wallet.hdwallet.accounts[0]._address_labels_backup) {
+        console.info('Migrate address labels from wallet to KV-Store v1.0.0');
+      } else {
+        // This is a new wallet
+      }
+
+      if (!this._wallet.isMetadataReady) {
+        // 2nd password is enabled and we can't write to the KV store
+        this._readOnly = true;
+      }
+
+      object = {
+        version: '1.0.0',
+        accounts: []
+      };
+
+      for (let account of this._wallet.hdwallet.accounts) {
+        let labels = [];
+        for (let label of account._address_labels_backup || []) {
+          labels[label.index] = {label: label.label};
+        }
+        // Set undefined entries to null:
+        for (let i = 0; i < labels.length; i++) {
+          if (!labels[i]) {
+            labels[i] = null;
+          }
+        }
+        object.accounts.push(labels);
+      }
+    } else if (major > 1) {
+      // Payload contains unsuppored new major version, abort:
+      throw new Error('LABELS_UNSUPPORTED_MAJOR_VERSION');
+    } else if (major === 1 && minor > 0) {
+      // New minor version can safely be used in read-only mode:
+      this._readOnly = true;
+    } else if (major === 1 && minor === 0 && patch > 0) {
+      // New patch version can safely to be used.
+    }
+
+    // Run (future) migration scripts:
+    // switch (object.version) {
+    //   case '1.0.0':
+    //     // Migrate from 1.0.1 to e.g. 1.0.2 or 1.1.0:
+    //     // ...
+    //     object.version = '1.0.1';
+    //     // falls through
+    //   case '1.0.1':
+    //     // Migrate from 1.0.1 to e.g. 1.0.2 or 1.1.0:
+    //     // ...
+    //     object.version = '1.0.2';
+    //     // falls through
+    //   default:
+    // }
+    return object;
+  }
+
+  // Goes through all labeled addresses and checks which ones have transactions.
+  // This result will be cached in the future. Although we obtain the balance,
+  // this is an implementation detail and we don't save it.
+  checkIfUsed (accountIndex) {
+    assert(Helpers.isPositiveInteger(accountIndex), 'specify accountIndex');
+    let labeledAddresses = this.all(accountIndex).filter((a) => a !== null);
+    let addresses = labeledAddresses.map((a) => a.address);
+
+    if (addresses.length === 0) return Promise.resolve();
+
+    BlockchainAPI.getBalances(addresses).then((data) => {
+      for (let labeledAddress of labeledAddresses) {
+        if (data[labeledAddress.address]) {
+          labeledAddress.used = data[labeledAddress.address].n_tx > 0;
+        }
+      }
+    });
+
+    return Promise.resolve();
```

Comment:
Should this function be returning the result of `getBalances`? Seems misleading to resolve before the function is done with its task.